### ISSUE TITLE
Docker build output tar file not working

### ISSUE THREAD

**Issue by SpikePy** [docker/cli#2680]
**Description**

If you want to build a tar file of an image with `docker build --output type=tar,dest=out.tar .` like described here: https://docs.docker.com/engine/reference/commandline/build/#custom-build-outputs it just does not work. there is no file created! Also the `man docker-build` doesn't inclode information about the --output flag.

**Steps to reproduce the issue:**
1. run `docker build --output type=tar,dest=out.tar .` in a directory with a **Dockerfile**

**Describe the results you received:**
 out.tar is not created

**Describe the results you expected:**
 out.tar is created

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:45:50 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a66213fe
  Built:            Mon Jun 22 15:44:21 2020
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false

Server:
 Containers: 2
  Running: 2
  Paused: 0
  Stopped: 0
 Images: 28
 Server Version: 19.03.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.7.0-2-amd64
 Operating System: Debian GNU/Linux bullseye/sid
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.779GiB
 Name: work
 ID: EE5A:L6DH:PNMV:D7H4:BKJD:MBNN:KF4T:F5UW:PHZY:KCDZ:3GWH:QFUR
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by iggyvolz** [comment]
I also have this issue - setting `DOCKER_BUILDKIT=1` or otherwise enabling buildkit (https://docs.docker.com/develop/develop-images/build_enhancements/) fixes the issue, but documentation should be updated to reflect that this is required.

**Comment by Myridium** [comment]
Setting `DOCKER_BUILDKIT=1` does not resolve this issue. The program should throw an error explaining the problem.

**Comment by Myridium** [comment]
One needs to `export DOCKER_BUILDKIT=1`. Of course there is no output indicating that something is wrong; the program brazenly proceeds, pretending that `--output` was never provided I guess, and compiles the image to the system where I did not want it.

**Comment by maximillianfx** [comment]
Yes @Myridium, but the docs already says about the need of `DOCKER_BUILDKIT` be exported to the env. You can export the `DOCKER_BUILDKIT` or use this flag in the beginning of the command:

`DOCKER_BUILDKIT=1 docker build --output type=tar,dest=file.tar .`

**Comment by dragonbook** [comment]
When use sudo,
sudo DOCKER_BUILDKIT=1 docker build --output type=tar,dest=file.tar .     **works**.
DOCKER_BUILDKIT=1 sudo docker build --output type=tar,dest=file.tar .     **not working**.

--
ubuntu 20.04
Docker version 20.10.3, build 48d30b5

**Comment by Bjohnson131** [comment]
I'm confused - why would someone want to `docker build` without the buildkit?

**Comment by satuday** [comment]
just spend a whole day debugging this issue. docker should throw a warning or error when DOCKER_BUILDKIT needs to be enable for --output. just ignoring the --output and proceed like everything is working is setting people up for failure.

**Comment by thaJeztah** [comment]
> just spend a whole day debugging this issue. docker should throw a warning or error when DOCKER_BUILDKIT needs to be enable for --output. just ignoring the --output and proceed like everything is working is setting people up for failure.

Yes, https://github.com/docker/cli/pull/2736 should fix that (but isn't merged yet)
